infra/aws/main.tf

```diff
@@ -52,8 +52,27 @@ module "aws_k8s" {
   vpc_id       = module.aws_base[0].vpc_id
   vpc_subnets  = module.aws_base[0].vpc_subnets
   name_prefix  = "${var.name_prefix}-${var.cluster_id}-${random_string.random_id.result}"
-  cluster_name = var.cluster_name == null ? "eks-${var.aws_k8s_region}-${var.name_prefix}" : var.cluster_name
+  cluster_name = coalesce(var.cluster_name, "eks-${var.aws_k8s_region}-${var.name_prefix}")
   output_path  = var.output_path
   tags         = local.default_tags
   depends_on   = [module.aws_jumpbox[0]]
 }
+
+module "external_dns" {
+  source                     = "../../modules/addons/aws/external-dns"
+  name_prefix                = "${var.name_prefix}-${var.cluster_id}"
+  cluster_name               = coalesce(var.cluster_name, "eks-${var.aws_k8s_region}-${var.name_prefix}")
+  k8s_host                   = module.aws_k8s[0].host
+  k8s_cluster_ca_certificate = module.aws_k8s[0].cluster_ca_certificate
+  k8s_client_token           = module.aws_k8s[0].token
+  region                     = var.aws_k8s_region
+  vpc_id                     = module.aws_base[0].vpc_id
+  dns_zone                   = var.external_dns_aws_dns_zone
+  sources                    = var.external_dns_sources
+  annotation_filter          = var.external_dns_annotation_filter
+  label_filter               = var.external_dns_label_filter
+  interval                   = var.external_dns_interval
+  tags                       = local.default_tags
+  oidc_provider_arn          = module.aws_k8s[0].oidc_provider_arn
+  external_dns_enabled       = var.external_dns_enabled
```

Comment:
Here and in the others, what about not propagating the `enabled` flag inside, and just use it as a `count` in this root module declaration? You control here at once if the module is applied or not, instead of having to "count" on every single resource inside the extdns module.

Reply:
as we do have helm provider configuration under the module... https://github.com/tetrateio/tetrate-service-bridge-sandbox/blob/6cc9d2410b3f7f7944665733fa9d5768aa996860/modules/addons/azure/external-dns/main.tf#L47 

in a way it goes to the point of externalizing... provider configuration away from module...

```
╷
│ Error: Module is incompatible with count, for_each, and depends_on
│
│   on main.tf line 55, in module "external_dns":
│   55:   count                      =  var.external_dns_enabled == true ? 1 : 0
│
│ The module at module.external_dns is a legacy module which contains its own local provider configurations, and so calls to it may not use the count, for_each, or depends_on arguments.
│
│ If you also control the module "../../modules/addons/azure/external-dns", consider updating this module to instead expect provider configurations to be passed by its caller.
╵
╷
```

Reply:
ok, not a big deal then